test/core/api/transport/listener_test.hpp

```diff
@@ -102,10 +106,12 @@ struct ListenerTest : public ::testing::Test {
   using SessionPtr = std::shared_ptr<Session>;
   using SubscriptionEngineType =
       SubscriptionEngine<Buffer, SessionPtr, Buffer, BlockHash>;
-  std::shared_ptr<SubscriptionEngineType> subscription_engine =
-      std::make_shared<SubscriptionEngineType>();
-  EventsSubscriptionEnginePtr events_engine =
-      std::make_shared<EventsSubscriptionEngineType>();
+  std::shared_ptr<StorageSubscriptionEngine> storage_events_engine =
```

Comment:
To according to other cases
```suggestion
  StorageSubscriptionEnginePtr storage_events_engine =
```